ecosystem/sep-0024.md

```diff
@@ -426,13 +426,13 @@ As an alternative to using the `callback` or `on_change_callback` parameters, th
 
 #### Guidance for anchors: closing the interactive popup
 
-After the user has provided all necessary information to the anchor through the interactive popup (`url`), the anchor should either close the popup automatically or instruct the user to close it themselves. If the anchor needs to provide instructions to the user for sending funds to it's off-chain account (in the deposit case), the anchor should not close the popup until the user has indicated to do so.
+After the user has provided all necessary information to the anchor through the interactive popup (`url`), the anchor should instruct the user to close the popup. This is necessary for deposit flows because the user must read and understand the instructions provided by the anchor for sending deposit funds. While this is not the case for withdrawal flows, anchors cannot assume the wallet will close the popup in a timely manner after updating the transaction's status.
```

Comment:
💡 I think we should write this to be more specific. It occurs to me reading this that it is in a user's best interest for the wallet not to try to close the interactive flow, even if the wallet has received some event that indicates the flow is complete. As you point out the deposit flow will display meaningful information to the user. In a similar way it could be jarring for the user for the withdrawal flow to disappear on its own.

```suggestion
After the user has provided all necessary information to the anchor through the interactive popup (`url`), the anchor should instruct the user to close the popup. The anchor or wallet should not close the interactive popup if they detect the payment is complete, so that the user has ample time to read the final deposit instructions, or in the case of withdrawals, has ample time to read any information the anchor displays at the end of the withdrawal flow.
```

I know we've discussed recently how closing automatically could be a better UX, but I also think that could be a more confusing UX. Since we already need no auto-close on deposit it is simpler to keep both deposit and withdrawal the same, and I suspect it will be less jarring as well.

Thoughts?

Reply:
For sure, auto-closing (i.e. not initiated by user action) is bad UX in any case.

However, prior to deprecating the `postMessage` callback, anchors could provide a 'Close' button that would trigger a postMessage call when clicked by the user, and the wallet would consequently close the window. This isn't jarring because the action is user-initiated, and could be used for both deposit and withdraw flows. 

This is the ideal UX, but it depends on anchors being able to make a postMessage call. Depending on the results of our internal discussions, we may end up un-deprecating postMessage to enable this UX.

If we decide to continue on this route (deprecating postMessage) then I agree with the change is phrasing here 👍